core/blockinfo/service.go

```diff
@@ -0,0 +1,50 @@
+package blockinfo
+
+import "time"
+
+type Service interface {
+	GetHeight() int64           // GetHeight returns the height of the block
+	Misbehavior() []Misbehavior // Misbehavior returns the misbehavior of the block
+	GetHeaderHash() []byte      // GetHeaderHash returns the hash of the block header
+	// GetValidatorsHash returns the hash of the validators
+	// For Comet, it is the hash of the next validator set
+	GetValidatorsHash() []byte
+	GetProposerAddress() []byte       // GetProposerAddress returns the address of the block proposer
+	GetDecidedLastCommit() CommitInfo // GetDecidedLastCommit returns the last commit info
+}
+
+// MisbehaviorType is the type of misbehavior for a validator
```

Comment:
would love feedback on defining our own types like this

Reply:
Seems reasonable to me.